argusclient/client.py

```diff
@@ -410,12 +440,80 @@ def get_user_dashboards(self, ownerName=None, shared=True, limit=None, version=N
 
 
+
+class GroupPermissionsServiceClient(BaseUpdatableModelServiceClient):
+    """
+    Service class that interfaces with the Argus grouppermissions endpoint.
+
+    There is no need to instantiate this directly, as it is available as :attr:`argusclient.client.ArgusServiceClient.grouppermissions` attribute.
+    """
+
+    def __init__(self, argus, get_all_req_opts=None):
+        """
+      :param get_all_req_opts: See BaseModelServiceClient.__init__() for description.
+      """
+        if not get_all_req_opts:
+            get_all_req_opts = {}
+        get_all_req_opts.setdefault(REQ_PATH, "grouppermission")
+        print("init happening")
+        super(GroupPermissionsServiceClient, self).__init__(Permission, argus, id_path="grouppermission/",
+                                                            get_all_req_opts=get_all_req_opts)
+
+    def _init_all(self, coll=None):
+        if not self.get_all_req_opts.get(REQ_PATH):
+            raise TypeError("Unsupported operation on: %s" % type(self))
+        if not self._retrieved_all:
+            resp = convert(self.argus._request(self.get_all_req_opts.get(REQ_METHOD, "get"),
+                                               self.get_all_req_opts.get(REQ_PATH, None),
+                                               params=self.get_all_req_opts.get(REQ_PARAMS, None),
+                                               dataObj=self.get_all_req_opts.get(REQ_BODY, None)))
+            for group_id, perms in resp.items():
+                self._coll[group_id] = perms
+            self._retrieved_all = True
+
+
+    def get_permissions_for_group(self, groupId):
+        resp = convert(self.argus._request("get", "grouppermission", params=list(groupId)))
+        if resp:
+            for group_id, perms in resp.items():
+                self._coll[group_id] = perms
+        return self._coll
+
+    def add_permissions_for_group(self, grouppermission):
+        """
+               Updates permissions in groups
+               :return: a list of :class:`argusclient.model.GroupPermission` objects with all fields populated.
+        """
+        if not isinstance(grouppermission, GroupPermission):
+            raise TypeError("Need a GroupPermission object, got: %s" % type(grouppermission))
+
+        updatedGroup = convert(self.argus._request("post", "grouppermission", dataObj=grouppermission))
+        self._coll[updatedGroup.groupId] = updatedGroup.permissionIds
+        return updatedGroup
+
+    def delete_permissions_for_group(self, grouppermission):
+        deleted_permission = None
+        if not isinstance(grouppermission, Permission):
+            raise TypeError("Need a Permission object, got: %s" % type(grouppermission))
+        perms_to_delete = grouppermission.permissionIds
+        if perms_to_delete == []:
+            raise ValueError("Permission is not already assigned and hence cant be deleted")
+        if tuple(perms_to_delete) in self._coll:
```

Comment:
do we need this check? I think we should make the request anyway even if it is not in the collection